presto-spark-classloader-interface/src/main/java/com/facebook/presto/spark/classloader_interface/PrestoSparkShuffleSerializer.java

```diff
@@ -0,0 +1,289 @@
+        {
+            throw new UnsupportedOperationException("this method is never used by shuffle");
+        }
+
+        @Override
+        public <T> SerializationStream writeAll(Iterator<T> iterator, ClassTag<T> classTag)
+        {
+            throw new UnsupportedOperationException("this method is never used by shuffle");
+        }
+    }
+
+    public static class PrestoSparkShuffleDeserializationStream
+            extends DeserializationStream
+    {
+        private final DataInputStream inputStream;
+
+        private final MutablePartitionId mutablePartitionId = new MutablePartitionId();
+        private final PrestoSparkMutableRow row = new PrestoSparkMutableRow();
+        private final Tuple2<Object, Object> tuple = new Tuple2<>(mutablePartitionId, row);
+
+        private byte[] bytes;
+        private ByteBuffer buffer;
+
+        public PrestoSparkShuffleDeserializationStream(InputStream inputStream)
+        {
+            this.inputStream = new DataInputStream(requireNonNull(inputStream, "inputStream is null"));
+        }
+
+        @Override
+        public Iterator<Tuple2<Object, Object>> asKeyValueIterator()
+        {
+            return asScalaIterator(new java.util.Iterator<Tuple2<Object, Object>>()
+            {
+                private Tuple2<Object, Object> next;
+
+                @Override
+                public boolean hasNext()
+                {
+                    if (next == null) {
+                        next = tryComputeNext();
+                    }
+                    return next != null;
+                }
+
+                @Override
+                public Tuple2<Object, Object> next()
+                {
+                    if (next == null) {
+                        next = tryComputeNext();
+                    }
+                    if (next == null) {
+                        throw new NoSuchElementException();
+                    }
+                    Tuple2<Object, Object> result = next;
+                    next = null;
+                    return result;
+                }
+
+                private Tuple2<Object, Object> tryComputeNext()
+                {
+                    int length;
+                    try {
+                        length = inputStream.readInt();
+                    }
+                    catch (EOFException e) {
+                        return null;
+                    }
+                    catch (IOException e) {
+                        throw new UncheckedIOException(e);
+                    }
+
+                    try {
+                        readRowData(length);
+                    }
+                    catch (IOException e) {
+                        throw new UncheckedIOException(e);
+                    }
+                    return tuple;
+                }
+            });
+        }
+
+        @Override
+        public <T> T readKey(ClassTag<T> classTag)
+        {
+            return (T) DEFAULT_KEY;
```

Comment:
What's this default key?

Reply:
`readKey` and `readValue` methods are actually not used by the shuffle. Let me remove them.